src/client.cpp

```diff
@@ -624,14 +625,20 @@ HSClient* get_client(const char* str) {
  *                  a decimal number its decimal window id.
  * \return          Window id, or 0, if unconvertable
  */
-Window get_window(const char* str) {
+Window get_window(const std::string& str) {
     // managed window?
-    auto client = get_client(str);
+    auto client = get_client(str.c_str());
     if (client)
         return client->window_;
 
     // unmanaged window? try to convert from base 16 or base 10 at the same time
-    return std::stoul(str, nullptr, 0);
+    try {
```

Comment:
I suggest using `Converter<unsigned long>::parse()` here for consistency on how we parse string input. Alternatively, remove the two superfluous arguments to stoul(). You can also use `catch (...)` instead of the two catches.

Reply:
I've droped the two default arugments of `stoul()` but kept the (only) two catch statements.

Reply:
I guess from a Python programmer's point of view it's always better to catch with strict typing. But in C++, and for this kind of standard function, is there any reason to assume that it could throw anything other than what is documented? (Not a rhetorical question, tbh...)

I figure that in this case, the distinction between exceptions type is meant for when you need to know what exactly went wrong (which we don't), so I'd second the suggestion to use `catch (...)`. 

For what it's worth, catching `std::logic_error` would also work.

Reply:
If at some point, we add `noexcept` to the function signature, we know that we did not forget any exception. But if you both prefer `catch(...)` I can change this.